ALICE3/TableProducer/onTheFlyTOFPID.cxx

```diff
@@ -0,0 +1,274 @@
+
+    auto mcCollision = collision.mcCollision();
+    o2::math_utils::Point3D<float> mcPvPos{mcCollision.posX(), mcCollision.posY(), mcCollision.posZ()};
+    std::array<float, 6> mcPvCov; // dummy!
+    for (int ii = 0; ii < 6; ii++)
+      mcPvCov[ii] = 1e-6;
+    o2::dataformats::VertexBase mcpvvtx(mcPvPos, mcPvCov);
+
+    for (const auto& track : tracks) {
+      // first step: find precise arrival time (if any)
+      // --- convert track into perfect track
+      if (!track.has_mcParticle()) // should always be OK but check please
+        LOG(error) << "Oh no! No mcParticle label for this track! This shouldn't happen!";
+
+      o2::track::TrackParCov o2track;
+      auto mcParticle = track.mcParticle();
+      convertMCParticleToO2Track(mcParticle, o2track);
+
+      float lX_PV = -100, lX_iTOF = -100, lX_oTOF = -100, lThisTrackLength_iTOF = -1, lThisTrackLength_oTOF = -1;
+      if (o2track.propagateToDCA(mcpvvtx, dBz))
+        lX_PV = o2track.getX();
+      if (!o2track.getXatLabR(iTOFRadius, lX_iTOF, dBz, o2::track::DirOutward))
+        lX_iTOF = -100;
+      if (!o2track.getXatLabR(oTOFRadius, lX_oTOF, dBz, o2::track::DirOutward))
+        lX_oTOF = -100;
+      if (lX_PV > -99. && lX_iTOF > -99.)
+        lThisTrackLength_iTOF = trackLength(o2track, lX_PV, lX_iTOF, dBz);
+      if (lX_PV > -99. && lX_oTOF > -99.)
+        lThisTrackLength_oTOF = trackLength(o2track, lX_PV, lX_oTOF, dBz);
+
+      // get mass to calculate velocity
+      auto pdgInfo = pdg->GetParticle(mcParticle.pdgCode());
+      float lExpectedTime_iTOF = lThisTrackLength_iTOF / Velocity(o2track.getP(), pdgInfo->Mass());
+      float lExpectedTime_oTOF = lThisTrackLength_oTOF / Velocity(o2track.getP(), pdgInfo->Mass());
+
+      // Smear with expected resolutions
+      float lMeasuredTime_iTOF = lPRNG.Gaus(lExpectedTime_iTOF, iTOFTimeReso);
+      float lMeasuredTime_oTOF = lPRNG.Gaus(lExpectedTime_oTOF, iTOFTimeReso);
+
+      // Now we calculate the expected arrival time following certain mass hypotheses
+      // and the (imperfect!) reconstructed track parametrizations
+      float lThisTrackLengthReco_iTOF = -1, lThisTrackLengthReco_oTOF = -1;
+      auto recoTrack = getTrackParCov(track);
+      if (recoTrack.propagateToDCA(pvvtx, dBz))
+        lX_PV = recoTrack.getX();
+      if (!recoTrack.getXatLabR(iTOFRadius, lX_iTOF, dBz, o2::track::DirOutward))
+        lX_iTOF = -100;
+      if (!recoTrack.getXatLabR(oTOFRadius, lX_oTOF, dBz, o2::track::DirOutward))
+        lX_oTOF = -100;
+      if (lX_PV > -99. && lX_iTOF > -99.)
+        lThisTrackLengthReco_iTOF = trackLength(recoTrack, lX_PV, lX_iTOF, dBz);
+      if (lX_PV > -99. && lX_oTOF > -99.)
+        lThisTrackLengthReco_oTOF = trackLength(recoTrack, lX_PV, lX_oTOF, dBz);
+
+      // Straight to Nsigma
+      float lDeltaTime_iTOF[5], lNSigma_iTOF[5];
+      float lDeltaTime_oTOF[5], lNSigma_oTOF[5];
+      int lpdg_array[5] = {11, 13, 211, 321, 2212};
```

Comment:
Don't use [magic numbers](https://rawgit.com/AliceO2Group/CodingGuidelines/master/coding_guidelines.html?showone=Magic_numbers#Magic_numbers). Use the enum particle names.